packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts

```diff
@@ -48,397 +28,163 @@ const supportedCurrencies: Record<string,  Record<RequestLogicTypes.CURRENCY, st
-      network: extensionAction.parameters.network,
-      acceptedTokens: extensionAction.parameters.acceptedTokens,
-      maxRateTimespan: extensionAction.parameters.maxRateTimespan,
-    },
-    version: extensionAction.version,
-  };
-}
-
-/**
- * Applies an add fee address and amount extension action
- *
- * @param extensionState previous state of the extension
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- * @param timestamp action timestamp
- *
- * @returns state of the extension updated
- */
-function applyAddFee(
-  extensionState: ExtensionTypes.IState,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): ExtensionTypes.IState {
-  if (
-    extensionAction.parameters.feeAddress &&
-    !isValidAddress(extensionAction.parameters.feeAddress)
-  ) {
-    throw Error('feeAddress is not a valid address');
-  }
-  if (extensionState.values.feeAddress) {
-    throw Error(`Fee address already given`);
-  }
-  if (
-    extensionAction.parameters.feeAmount &&
-    !Utils.amount.isValid(extensionAction.parameters.feeAmount)
-  ) {
-    throw Error('feeAmount is not a valid amount');
-  }
-  if (extensionState.values.feeAmount) {
-    throw Error(`Fee amount already given`);
-  }
-  if (!requestState.payee) {
-    throw Error(`The request must have a payee`);
-  }
-  if (!Utils.identity.areEqual(actionSigner, requestState.payee)) {
-    throw Error(`The signer must be the payee`);
-  }
-
-  const copiedExtensionState: ExtensionTypes.IState = Utils.deepCopy(extensionState);
-
-  // update fee address and amount
-  copiedExtensionState.values.feeAddress = extensionAction.parameters.feeAddress;
-  copiedExtensionState.values.feeAmount = extensionAction.parameters.feeAmount;
-
-  // update events
-  copiedExtensionState.events.push({
-    name: ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_FEE,
-    parameters: {
-      feeAddress: extensionAction.parameters.feeAddress,
-      feeAmount: extensionAction.parameters.feeAmount,
-    },
-    timestamp,
-  });
-
-  return copiedExtensionState;
-}
-
-/**
- * Check if an ethereum address is valid
- *
- * @param {string} address address to check
- * @returns {boolean} true if address is valid
- */
-function isValidAddress(address: string): boolean {
-  return walletAddressValidator.validate(address, 'ethereum');
-}
+    };
+  }
+
+  /**
+   * Validates the payment network of the request currency.
+   *
+   * @param currency
+   */
+  protected validateSupportedCurrency(
+    request: RequestLogicTypes.IRequest,
+    extensionAction: ExtensionTypes.IAction,
+  ): void {
+    const network =
+      extensionAction.parameters.network ||
+      request.extensions[this.paymentNetworkId]?.values.network;
+
+    // Nothing can be validated if the network has not been given yet
+    if (!network) {
+      return;
+    }
```

Comment:
That's a (potentially big) functional change, but I think it's valid. 

The former logic was to give a default network (mainnet) but it makes a difference between mainnet & rinkeby behaviour.